src/Workspaces/Remote/ServiceHub/Services/ProcessTelemetry/RemoteProcessTelemetryService.cs

```diff
@@ -43,6 +43,13 @@ public RemoteProcessTelemetryService(ServiceConstructionArguments arguments)
         {
         }
 
+        public override void Dispose()
+        {
+            _telemetryService?.Dispose();
```

Comment:
❗ This is a double-dispose and should be removed.

Reply:
The containing instance is created and owned by the MEF container. MEF will dispose it when the container is disposed.

Reply:
Good catch! Fixed in commit 2.

Reply:
As our servicehub MEF container isn't being disposed, this actually isn't a double dispose and is needed to ensure the aggregated telemetry events are posted. I understand the concern of slowing down process shutdown, but the TelemetrySession documentation alleviates that concern somewhat by indicating "all methods return immediately" and events are "saved locally if they cannot be sent before the process exits".

If the consensus is to not post the telemetry events on shutdown, we can remove this code and just depend on the timeout threshholds to send the aggregated data.